ycmd/completers/all/identifier_completer.py

```diff
@@ -37,6 +37,10 @@ def __init__( self, user_options ):
     self._completer = ycm_core.IdentifierCompleter()
     self._tags_file_last_mtime = defaultdict( int )
     self._logger = logging.getLogger( __name__ )
+    try:
+      self._max_candidates = user_options[ 'max_num_identifier_candidates' ]
+    except KeyError:
+      self._max_candidates = MAX_IDENTIFIER_COMPLETIONS_RETURNED
```

Comment:
@ptrv Could you add a bit of code to merge the provided options with the default options? Basically have the provided options override the default ones, but still have a value for every field.